### PR TITLE
feat: Add more tests to `ecocredit` package

### PR DESCRIPTION
**Pull request by atheeshp**
## Description

Closes: #423 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/msgs.go [hunk 1]
```diff
@@ -13,8 +13,20 @@ var (
...
 
 func (m *MsgCreateClass) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Designer)
+	if err != nil {
+		return sdkerrors.Wrap(err, "designer")
+	}
+
 	if len(m.Issuers) == 0 {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "issuers cannot be empty")
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Issuers cannot be empty")
```
Comment (blushi) [inline x/ecocredit/msgs.go:22]: I believe we should keep it lower case

File: x/ecocredit/msgs_test.go [hunk 2]
```diff
@@ -0,0 +1,741 @@
+package ecocredit
```
Comment (blushi) [inline x/ecocredit/msgs_test.go:1]: could you add tests for `MsgSetPrecision`?

**Review by blushi — CHANGES_REQUESTED** [review]
Thanks @atheeshp, looks good overall!
Now that we have unit tests for msgs `ValidateBasic()` methods, I think we can safely remove  all the `ValidateBasic()` tests in server/testsuite/suite.go as long as those test cases are covered in msgs_test.go

File: x/ecocredit/msgs_test.go (hunk 2 above)
Comment (ruhatch) [reply x/ecocredit/msgs_test.go:1]: I think we're going to remove `MsgSetPrecision` as part of #428, so might not be a good use of time to add them. What do you think @blushi?

File: x/ecocredit/server/msg_server.go [hunk 3]
```diff
@@ -26,6 +26,13 @@ func (s serverImpl) CreateClass(goCtx context.Context, req *ecocredit.MsgCreateC
 	classID := s.idSeq.NextVal(ctx)
 	classIDStr := util.Uint64ToBase58Check(classID)
 
+	for _, issuer := range req.Issuers {
```
Comment (ruhatch) [inline x/ecocredit/server/msg_server.go:29]: I guess we don't need this here if it's in `ValidateBasic()`, right?

File: x/ecocredit/msgs_test.go (hunk 2 above)
Comment (blushi) [reply x/ecocredit/msgs_test.go:1]: makes sense!

**Review by blushi — APPROVED** [review]

File: x/ecocredit/msgs.go [hunk 4]
```diff
@@ -30,22 +42,35 @@ func (m *MsgCreateClass) GetSigners() []sdk.AccAddress {
 }
 
 func (m *MsgCreateBatch) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Issuer)
+	if err != nil {
+		return err
```
Comment (anilCSE) [inline x/ecocredit/msgs.go:47]: let's be consistent
```suggestion
		return sdkerrors.Wrap(err, "issuer")
```

**Review by anilCSE — APPROVED** [review]
lgtm, nits to keep consistency with address validation errors

**Pull request merged**